WalletWasabi/Services/WalletService.cs

```diff
@@ -487,7 +487,7 @@ private void ProcessTransaction(SmartTransaction tx, List<HdPubKey> keys = null)
 			{
 				// If transaction received to any of the wallet keys:
 				var output = tx.Transaction.Outputs[i];
-				HdPubKey foundKey = keys.SingleOrDefault(x => x.GetP2wpkhScript() == output.ScriptPubKey);
```

Comment:
Isn't the only improvement is this lookup? Don't we get the same performance if we simply replace this `SingleOrDefault` with `FirstOrDefaults`?  

Since, if you look into `HdPubKey.GetP2wpkhScript()`, then you'd recognize it's written in a really performant way.

Reply:
No, it's not. I tested doing a lot of things:

Currently - 53sec
Removing the nullchecks from HdPubKey - 53sec
Refactor HdPubKey to use autoproperties - 53sec
Use FirstOrDefault instead SingleOrDefault - 53sec
Use Contains instead SingleOrDefault - 53sec
Use Any instead SingleOrDefault - 53sec
Lucas PR - 44sec